spec.md

```diff
@@ -1663,6 +1717,514 @@ message GetCapacityResponse {
+
+| Condition | gRPC Code | Description | Recovery Behavior |
+|-----------|-----------|-------------|-------------------|
+| Invalid `starting_token` | 10 ABORTED | Indicates that `starting_token` is not valid. | Caller SHOULD start the `ListVolumeGroups` operation again with an empty `starting_token`. |
+
+#### `CreateVolumeGroupSnapshot`
+
+**ALPHA FEATURE**
+
+The purpose of this call is to request the creation of a multi-volume snapshot. Group snapshots can be created from existing volume group. Note that calls to this function must be idempotent - the function may be called multiple times for the same name - the group snapshot must only be created once.
+
+In VolumeGroupSnapshot message, snapshots is an optional field while group_snapshot_id is a required field. It is fine to only specify group_snapshot_id but not snapshots in VolumeGroupSnapshot message at restore time. However, the Plugin MUST return a list of volumes that are restored in CreateVolumeGroupResponse.
+
+```protobuf
+message CreateVolumeGroupSnapshotRequest {
+  option (alpha_message) = true;
+
+  // suggested name for a group snapshot (required for idempotent)
+  // This field is REQUIRED.
+  string name = 1;
+
+  // identifier indicates which volume group is used to take
+  // group snapshot
+  // This field is REQUIRED.
+  string source_volume_group_id = 2;
+
+  // volume ids of the volumes in the source group. This field is
+  // REQUIRED. This is needed because some storage systems does not
+  // have a group persisted on the storage system until the time to
+  // take a group snapshot
+  repeated string volume_ids = 3;
+
+  // secrets required for snapshot creation (pulled from
+  // VolumeSnapshotClass)
+  // This field is OPTIONAL.
+  map<string, string> secrets = 4 [(.csi.v1.csi_secret) = true];
+
+  // params passed from VolumeSnapshotClass
+  // This field is OPTIONAL.
+  map<string, string> parameters = 5;
+}
+
+message CreateVolumeGroupSnapshotResponse {
+  option (alpha_message) = true;
+
+  // Contains all attributes of the newly created group snapshot.
+  // This field is REQUIRED.
+  VolumeGroupSnapshot group_snapshot = 1;
+}
+
+message VolumeGroupSnapshot {
+  option (alpha_message) = true;
+
+  // The identifier for this group snapshot, generated by the plugin.
+  // This field is REQUIRED.
+  string group_snapshot_id = 1;
+
+  // A list of snapshots created. Snapshot is the same
+  // definition as Snapshot definition used in CSI.
+  // This field is OPTIONAL.
+  repeated .csi.v1.Snapshot snapshots = 2;
+
+  // Identity information for the source volume group. Currently, only
+  // support the case that source is volume group. This field is
+  // REQUIRED.
+  string source_volume_group_id = 3;
+
+  // Indicates if a list of group snapshots are ready.
+  // This field is REQUIRED.
+  bool ready_to_use = 4;
+
+  // Timestamp when the point-in-time consistency group snapshot is
```

Comment:
do all backends support this? if a backend tracks time in "ticks" or some vector clock unit that doesn't translate well to timestamp .. when then? why does the CO care about this field?

Reply:
I'll remove the "consistency" part but keep the creation_time.  We have timestamp for individual snapshots already.